### PR TITLE
Bug/attribute amino cdc

### PR DESCRIPTION
**Pull request by arnabmitra**
## Description

closes: #188 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: x/attribute/keeper/keeper.go [hunk 1]
```diff
@@ -78,20 +79,24 @@ func (k Keeper) GetAttributes(ctx sdk.Context, acc sdk.AccAddress, name string)
...
-	iterator := sdk.KVStorePrefixIterator(store, types.AttributeKeyPrefix)
+	iterator := sdk.KVStorePrefixIterator(store, prefix)
 	defer iterator.Close()
 	// Iterate over records, processing callbacks.
 	for ; iterator.Valid(); iterator.Next() {
 		record := types.Attribute{}
-		if err := types.ModuleCdc.UnmarshalBinaryBare(iterator.Value(), &record); err != nil {
-			return err
+		// get proto objects for legacy prefix with legacy amino codec.
+		if bytes.Equal(types.AttributeKeyPrefix, prefix) {
```
Comment (iramiller) [inline x/attribute/keeper/keeper.go:91]: Following line is using amino, probably want amino prefix here

File: x/attribute/keeper/keeper.go [hunk 2]
```diff
@@ -78,20 +79,24 @@ func (k Keeper) GetAttributes(ctx sdk.Context, acc sdk.AccAddress, name string)
 }
 
 // IterateRecords iterates over all the stored attribute records and passes them to a callback function.
-func (k Keeper) IterateRecords(ctx sdk.Context, handle Handler) error {
+func (k Keeper) IterateRecords(ctx sdk.Context, prefix []byte, handle Handler) error {
```
Comment (iramiller) [inline x/attribute/keeper/keeper.go:82]: Having this prefix param here kind of leaks some implementation details out of the keeper that I am not sure we really should...

File: x/attribute/keeper/keeper.go (hunk 1 above)
Comment (arnabmitra) [reply x/attribute/keeper/keeper.go]: good catch! :( changed

File: x/attribute/keeper/keeper.go (hunk 2 above)
Comment (arnabmitra) [reply x/attribute/keeper/keeper.go:82]: we can remove this after we do the conversion, but technically an iterate handler could be using to iterate over different prefixes(rather keys, like the name module) without writing methods for each, i thought having a generic handler already violates the encapsulation principle here, meaning that the caller is responsible for how/what they iterate/filter through..but for this one we can remove the param after this conversion..or we can write the a custom method for that prefix(dup code, but again can be removed)..

**Review by iramiller — APPROVED** [review]

**Pull request merged**